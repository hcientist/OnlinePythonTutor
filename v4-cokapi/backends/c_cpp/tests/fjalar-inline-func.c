// From the test suite of https://github.com/codespecs/daikon
//   daikon/tests/kvasir-tests/

int foo(int);

inline int foo(int x) {
  return 5;
}

int main() {
  foo(3);
  return 0;
}
