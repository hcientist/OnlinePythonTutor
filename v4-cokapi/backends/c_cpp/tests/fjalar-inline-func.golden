{
  "code": "// From the test suite of https://github.com/codespecs/daikon\n//   daikon/tests/kvasir-tests/\n\nint foo(int);\n\ninline int foo(int x) {\n  return 5;\n}\n\nint main() {\n  foo(3);\n  return 0;\n}", 
  "trace": [
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 11, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFF0001D0", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 10, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF0001D0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "call", 
      "func_name": "foo", 
      "globals": {}, 
      "heap": {}, 
      "line": 6, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFF0001D0", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 11, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF0001D0"
        }, 
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF0001BC", 
              "int", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF0001C0", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 6, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF0001C0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "foo", 
      "globals": {}, 
      "heap": {}, 
      "line": 7, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFF0001D0", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 11, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF0001D0"
        }, 
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF0001BC", 
              "int", 
              3
            ]
          }, 
          "frame_id": "0xFFF0001C0", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 6, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF0001C0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "foo", 
      "globals": {}, 
      "heap": {}, 
      "line": 8, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFF0001D0", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 11, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF0001D0"
        }, 
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF0001BC", 
              "int", 
              3
            ]
          }, 
          "frame_id": "0xFFF0001C0", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 8, 
          "ordered_varnames": [
            "x"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF0001C0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 12, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFF0001D0", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 12, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF0001D0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 13, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFF0001D0", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 13, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF0001D0"
        }
      ], 
      "stdout": ""
    }
  ]
}

